crates/mirro-rs/man/mirro-rs.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH mirro-rs 1  "mirro-rs 0.1.0-alpha.1" 
.SH NAME
mirro\-rs \- An ArchLinux mirrorlist manager with a TUI
.SH SYNOPSIS
\fBmirro\-rs\fR [\fB\-o\fR|\fB\-\-outfile\fR] [\fB\-e\fR|\fB\-\-export\fR] [\fB\-v\fR|\fB\-\-view\fR] [\fB\-s\fR|\fB\-\-sort\fR] [\fB\-t\fR|\fB\-\-ttl\fR] [\fB\-u\fR|\fB\-\-url\fR] [\fB\-r\fR|\fB\-\-rate\fR] [\fB\-\-timeout\fR] [\fB\-i\fR|\fB\-\-include\fR] [\fB\-d\fR|\fB\-\-direct\fR] [\fB\-a\fR|\fB\-\-age\fR] [\fB\-c \fR] [\fB\-p\fR|\fB\-\-protocols\fR] [\fB\-\-ipv4\fR] [\fB\-\-ipv6\fR] [\fB\-\-isos\fR] [\fB\-\-completion\-percent\fR] [\fB\-h\fR|\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] 
.SH DESCRIPTION
An ArchLinux mirrorlist manager with a TUI
.SH OPTIONS
.TP
\fB\-o\fR, \fB\-\-outfile\fR=\fIOUTFILE\fR
File to write mirrors to
.TP
\fB\-e\fR, \fB\-\-export\fR=\fIEXPORT\fR
Number of mirrors to export [default: 50]
.TP
\fB\-v\fR, \fB\-\-view\fR=\fIVIEW\fR
An order to view all countries
.br

.br
[\fIpossible values: \fRalphabetical, mirror\-count]
.TP
\fB\-s\fR, \fB\-\-sort\fR=\fISORT\fR
Default sort for exported mirrors
.br

.br
[\fIpossible values: \fRpercentage, delay, duration, score]
.TP
\fB\-t\fR, \fB\-\-ttl\fR=\fITTL\fR
Number of hours to cache mirrorlist for
.TP
\fB\-u\fR, \fB\-\-url\fR=\fIURL\fR
URL to check for mirrors
.TP
\fB\-r\fR, \fB\-\-rate\fR=\fIRATE\fR
Sort mirrorlists by download speed when exporting
.TP
\fB\-\-timeout\fR=\fITIMEOUT\fR
Connection timeout in seconds
.TP
\fB\-i\fR, \fB\-\-include\fR=\fIINCLUDE\fR
Extra CDNs to check for mirrors
.TP
\fB\-d\fR, \fB\-\-direct\fR=\fIDIRECT\fR
Skip TUI session and directly export the mirrorlist
.TP
\fB\-a\fR, \fB\-\-age\fR=\fIAGE\fR
How old (in hours) should the mirrors be since last synchronisation
.TP
\fB\-c\fR=\fICOUNTRY\fR
Countries to search for mirrorlists
.TP
\fB\-p\fR, \fB\-\-protocols\fR=\fIPROTOCOLS\fR
Filters to use on mirrorlists
.br

.br
[\fIpossible values: \fRhttps, http, rsync]
.TP
\fB\-\-ipv4\fR=\fIIPV4\fR
Only return mirrors that support IPv4
.TP
\fB\-\-ipv6\fR=\fIIPV6\fR
Only return mirrors that support IPv6
.TP
\fB\-\-isos\fR=\fIISOS\fR
Only return mirrors that host ISOs
.TP
\fB\-\-completion\-percent\fR=\fICOMPLETION_PERCENT\fR
Set the minimum completion percent for the returned mirrors
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help information
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version information
.SH VERSION
v0.1.0\-alpha.1


crates/mirro-rs/completions/mirro-rs.bash
_mirro-rs() {
    local i cur prev opts cmds
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="mirro__rs"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        mirro__rs)
            opts="-o -e -v -s -t -u -r -i -d -a -c -p -h -V --outfile --export --view --sort --ttl --url --rate --timeout --include --direct --age --protocols --ipv4 --ipv6 --isos --completion-percent --help --version"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --outfile)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --export)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -e)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --view)
                    COMPREPLY=($(compgen -W "alphabetical mirror-count" -- "${cur}"))
                    return 0
                    ;;
                -v)
                    COMPREPLY=($(compgen -W "alphabetical mirror-count" -- "${cur}"))
                    return 0
                    ;;
                --sort)
                    COMPREPLY=($(compgen -W "percentage delay duration score" -- "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -W "percentage delay duration score" -- "${cur}"))
                    return 0
                    ;;
                --ttl)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -t)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --url)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -u)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --timeout)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --include)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -i)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --age)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -a)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -c)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --protocols)
                    COMPREPLY=($(compgen -W "https http rsync" -- "${cur}"))
                    return 0
                    ;;
                -p)
                    COMPREPLY=($(compgen -W "https http rsync" -- "${cur}"))
                    return 0
                    ;;
                --completion-percent)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

complete -F _mirro-rs -o bashdefault -o default mirro-rs
